server/src/test/java/org/elasticsearch/action/admin/cluster/settings/ClusterUpdateSettingsResponseTests.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.cluster.settings;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.settings.Settings.Builder;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.core.IsCollectionContaining.hasItem;
+
+public class ClusterUpdateSettingsResponseTests extends ESTestCase {
+
+    public void testFromToXContent() throws IOException {
+        final ClusterUpdateSettingsResponse response = createTestItem();
+        boolean humanReadable = randomBoolean();
+        final XContentType xContentType = XContentType.JSON;
+
+        BytesReference xContent = toShuffledXContentAndInsertRandomFields(response, xContentType, ToXContent.EMPTY_PARAMS, humanReadable);
+
+        XContentParser parser = createParser(xContentType.xContent(), xContent);
+        ClusterUpdateSettingsResponse parsedResponse = ClusterUpdateSettingsResponse.fromXContent(parser);
+
+        assertNull(parser.nextToken());
+        assertThat(parsedResponse.isAcknowledged(), equalTo(response.isAcknowledged()));
+
+        Builder persistentBuilder = Settings.builder().put(response.persistentSettings);
+        Builder parsedPersistentBuilder = Settings.builder().put(parsedResponse.persistentSettings);
+
+        persistentBuilder.keys().forEach(k -> {
+            assertThat(parsedPersistentBuilder.keys(), hasItem(equalTo(k)));
+            assertThat(parsedPersistentBuilder.get(k), equalTo(persistentBuilder.get(k)));
+        });
```

Comment:
trying to figure out what happens when random fields are inserted among settings. I would expect the test to fail somehow, or maybe the arbitrary fields get parsed as setting and that is why you don't compare the two maps directly?

I wonder if randomizing at this level should be disabled, as its goal would be to verify that we are forward compatible when parsing, meaning that we will ignore a field/object if we don't know about it, as it could be a field added in a more recent version.

Reply:
The `Settings` will be simply parsed ( no strict validation ) and during execution will be compared tp the supported settings. I will disable the randomization for the settings, then the check is only for the 'top-level' fields.